### PR TITLE
MA-27, 28: Create Client profile page (Dashboard and Bots tabs)

### PR DESCRIPTION
**Pull request by vitalii-petrun**
https://user-images.githubusercontent.com/87064627/213820221-c126fb0a-e4ee-4f06-bf6e-f5d017d32fdc.mp4

* version on video doesn't contain table which was  created later

### CODE REVIEW

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+          ]),
+        ),
+        SelectableText.rich(
+          textAlign: TextAlign.center,
+          TextSpan(children: [
+            TextSpan(
+              text: context.localizations.referralCodeForPartners,
+              style: _referralCodeTextStyle,
+            ),
+            const TextSpan(text: "  "),
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:257]: Ca we delete this and wrap previous text span in "$...   "?

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 2]
```diff
@@ -0,0 +1,755 @@
...
+            ),
+          ]),
+        ),
+        ColoredButton(
+          title: context.localizations.copyCode,
+          color: context.color.roadmapCardBackground,
+          onTap: () {
+            Clipboard.setData(const ClipboardData(text: _referralCode));
+            showOverlay(context);
+          },
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:270]: Pls move to separate method

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 3]
```diff
@@ -0,0 +1,755 @@
...
+    color: AppColors.white,
+  );
+  static final _partnerIncomeTextStyle = TextStyle(
+    fontSize: 14,
+    color: AppColors.white.withOpacity(0.6),
+  );
+  static const _levelTextStyle = TextStyle(
+    fontSize: 16,
+    color: AppColors.white,
+  );
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:338]: Same as above

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 4]
```diff
@@ -0,0 +1,755 @@
...
+  final _titleTextStyle = const TextStyle(
+    fontSize: 34,
+    color: AppColors.white,
+    fontWeight: FontWeight.w700,
+  );
+
+  final _labelTextStyle = TextStyle(
+    fontSize: 14,
+    color: AppColors.white.withOpacity(0.6),
+  );
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:396]: Same as above

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 5]
```diff
@@ -0,0 +1,755 @@
...
+              context.localizations.enterWalletAddress,
+              style: _labelTextStyle,
+            ),
+            const SizedBox(height: 6),
+            _FilledTextField(
+              hintText: "0x...",
+              controller: TextEditingController(),
+            ),
+            const SizedBox(height: 32),
+            Text(context.localizations.chooseNetwork, style: _labelTextStyle),
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:423]: Add comma

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 6]
```diff
@@ -0,0 +1,755 @@
...
+  final TextEditingController controller;
+  final String hintText;
+
+  const _FilledTextField({
+    required this.controller,
+    required this.hintText,
+    Key? key,
+  }) : super(key: key);
+
+  static const _hintTextStyle = TextStyle(fontSize: 16);
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:455]: Delete

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 7]
```diff
@@ -0,0 +1,755 @@
...
+  const _TransactionsHistorySection({Key? key}) : super(key: key);
+
+  static const _transactionsTextStyle = TextStyle(
+    fontSize: 20,
+  );
+  static const _timeTextStyle = TextStyle(
+    fontSize: 13,
+    color: AppColors.greyish,
+  );
+  static final _transactions = [
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:519]: Make const and not static

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 8]
```diff
@@ -0,0 +1,755 @@
...
+class _TransactionsHistorySection extends StatelessWidget {
+  const _TransactionsHistorySection({Key? key}) : super(key: key);
+
+  static const _transactionsTextStyle = TextStyle(
+    fontSize: 20,
+  );
+  static const _timeTextStyle = TextStyle(
+    fontSize: 13,
+    color: AppColors.greyish,
+  );
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:518]: Move

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 9]
```diff
@@ -0,0 +1,755 @@
...
+      '12 October 2020, 19:23',
+      'Cancelled',
+      'Deposit',
+    ),
+  ];
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      decoration: BoxDecoration(
+        color: AppColors.white,
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:546]: Styles

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 10]
```diff
@@ -0,0 +1,755 @@
...
+  }) : super(key: key);
+
+  TransactionStatus parse(String value) {
+    return TransactionStatus.values.firstWhere(
+      (element) => element.value == value,
+      orElse: () => TransactionStatus.pending,
+    );
+  }
+
+  static const _transactionTextStyle = TextStyle(
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:607]: Styles

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 11]
```diff
@@ -0,0 +1,755 @@
...
+      children: [
+        Row(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Container(
+              height: 50,
+              width: 50,
+              decoration: BoxDecoration(
+                borderRadius: BorderRadius.circular(100),
+                color: AppColors.black,
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:629]: Styles

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 12]
```diff
@@ -0,0 +1,755 @@
...
+                  transaction.network,
+                  style: _transactionTextStyle,
+                ),
+                const SizedBox(height: 6),
+                Text(
+                  context.localizations.paymentSystem,
+                  style: const TextStyle(
+                    fontSize: 13,
+                    color: AppColors.greyish,
+                  ),
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:646]: Styles

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 13]
```diff
@@ -0,0 +1,755 @@
...
+                transaction.date,
+                style: _transactionTextStyle,
+              ),
+              const SizedBox(height: 6),
+              Text(
+                context.localizations.operationDate,
+                style: const TextStyle(
+                  fontSize: 13,
+                  color: AppColors.greyish,
+                ),
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:667]: Styles

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 14]
```diff
@@ -0,0 +1,755 @@
...
+        return Colors.green;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final statusTextStyle = TextStyle(
+      fontSize: 13,
+      color: _chipColor,
+    );
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:718]: Inside build...

File: lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart [hunk 15]
```diff
@@ -0,0 +1,755 @@
...
+  final String status;
+  final String type;
+
+  Transaction(this.network, this.amount, this.date, this.status, this.type);
+}
+
+enum TransactionStatus {
+  completed("Completed"),
+  pending("Pending"),
+  canceled("Cancelled");
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/dashboard_tab.dart:751]: I think it should be localized

File: lib/presentation/pages/client_profile_page/sections/side_menu_section.dart [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+    Provider.of<ClientProfileManager>(context, listen: false)
+        .changeIndex(index);
+    setState(() => _currentIndex = index);
+  }
+
+  @override
+  void initState() {
+    _currentIndex =
+        Provider.of<ClientProfileManager>(context, listen: false).currentIndex;
+    super.initState();
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/side_menu_section.dart:28]: IninState in first place

File: lib/presentation/pages/client_profile_page/sections/side_menu_section.dart [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+          borderRadius: BorderRadius.circular(12),
+          child: Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: Row(
+              children: [
+                CircleAvatar(
+                  radius: 28,
+                  backgroundColor: _isSelected && menu.isCollapsed
+                      ? AppColors.white
+                      : AppColors.transparent,
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/side_menu_section.dart:107]: Styles

File: lib/presentation/pages/client_profile_page/sections/side_menu_section.dart [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+                  child: Icon(
+                    widget.icon,
+                    size: 34,
+                    color: _isSelected
+                        ? context.color.dashboardSideMenuSelectedItem
+                        : (menu.isCollapsed
+                            ? AppColors.darkIndigo
+                            : context.color.dashboardSideMenuUnselectedItem),
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/side_menu_section.dart:115]: Styles

File: lib/presentation/widgets/dashboard_header.dart [hunk 19]
```diff
@@ -0,0 +1,340 @@
...
+                      height: 68,
+                    )
+                  : const SizedBox(),
+              menu.isCollapsed
+                  ? const SizedBox()
+                  : SizedBox(width: context.screenWidth * 0.02),
+              IconButton(
+                splashColor: AppColors.transparent,
+                highlightColor: AppColors.transparent,
+                hoverColor: AppColors.transparent,
```
Comment (vovaklh) [inline lib/presentation/widgets/dashboard_header.dart:63]: Styles

File: lib/presentation/widgets/dashboard_header.dart [hunk 20]
```diff
@@ -0,0 +1,340 @@
...
+
+    _overlayEntry?.let((entry) {
+      WidgetsBinding.instance
+          .addPostFrameCallback((_) => overlayState?.insert(entry));
+      _animationController.forward();
+    });
+  }
+
+  removeOverlay() =>
+      _animationController.reverse().then((value) => _overlayEntry?.remove());
```
Comment (vovaklh) [inline lib/presentation/widgets/dashboard_header.dart:144]: Dont use then, use await

File: lib/presentation/widgets/dashboard_header.dart [hunk 21]
```diff
@@ -0,0 +1,340 @@
...
+  void didChangeDependencies() {
+    removeOverlay();
+    super.didChangeDependencies();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    const welcomeStyle = TextStyle(fontSize: 16);
+    const emailStyle = TextStyle(fontSize: 12, color: Color(0xFF464A53));
+    const userNameStyle = TextStyle(fontSize: 16, fontWeight: FontWeight.bold);
```
Comment (vovaklh) [inline lib/presentation/widgets/dashboard_header.dart:173]: Move

File: lib/presentation/widgets/dashboard_header.dart [hunk 22]
```diff
@@ -0,0 +1,340 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Material(
+      color: context.color.localeDropdownOpacity,
+      child: FadeTransition(
+        opacity: animation as Animation<double>,
+        child: Container(
+          decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(10),
+            color: AppColors.white,
```
Comment (vovaklh) [inline lib/presentation/widgets/dashboard_header.dart:225]: Move

**Review by vovaklh — CHANGES_REQUESTED** [review]

File: lib/presentation/pages/client_profile_page/sections/bots_tab.dart [hunk 23]
```diff
@@ -0,0 +1,537 @@
...
+  static List<BoxShadow> defaultShadow(BuildContext context) {
+    return [
+      BoxShadow(
+        color: context.color.primary.withOpacity(0.1),
+        blurRadius: 20,
+        offset: const Offset(0, 10),
+      ),
+    ];
+  }
+}
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/bots_tab.dart:114]: Pls delete

File: lib/presentation/pages/client_profile_page/sections/bots_tab.dart [hunk 24]
```diff
@@ -0,0 +1,537 @@
...
+                        color: context.color.secondaryVariant,
+                        borderRadius: BorderRadius.circular(30),
+                      ),
+                      padding: const EdgeInsets.symmetric(
+                        horizontal: 10,
+                        vertical: 5,
+                      ),
+                      child: Text(
+                        "ROI: $roi%",
+                        style: context.text.inverseBody.copyWith(fontSize: 11),
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/bots_tab.dart:202]: 12

File: lib/presentation/pages/client_profile_page/sections/bots_tab.dart [hunk 25]
```diff
@@ -0,0 +1,537 @@
...
+    );
+  }
+}
+
+class _FunctionalityComparisonTable extends StatelessWidget {
+  const _FunctionalityComparisonTable({Key? key}) : super(key: key);
+
+  TextStyle _defaultCellTextStyle(BuildContext context) {
+    return context.text.clientBotsDefault.copyWith(
+      fontSize: Responsive.isMobile(context) ? 14 : 17,
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/bots_tab.dart:317]: 18

File: lib/presentation/pages/client_profile_page/sections/bots_tab.dart [hunk 26]
```diff
@@ -0,0 +1,537 @@
...
+    );
+  }
+
+  static final _fourChecks = List.generate(
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/bots_tab.dart:321]: remove pls static

File: lib/presentation/pages/client_profile_page/sections/bots_tab.dart [hunk 27]
```diff
@@ -0,0 +1,537 @@
...
+    Key? key,
+  }) : super(key: key);
+
+  TextStyle _headerTableRowTextStyle(BuildContext context) {
+    final text = context.text;
+
+    return text.clientBotsDefault.copyWith(
+      color: text.clientBotsDefault.color?.withOpacity(0.5),
+      fontWeight: FontWeight.w200,
+      fontSize: Responsive.isMobile(context) ? 12 : 15,
```
Comment (vovaklh) [inline lib/presentation/pages/client_profile_page/sections/bots_tab.dart:504]: 16

File: lib/presentation/themes/src/constants.dart [hunk 28]
```diff
@@ -19,18 +19,27 @@ abstract class AppColors {
   static const lightGrey = Color(0xFFF9F8FA);
   static const darkerGrey = Color(0xFFDDDDDD);
   static const lighterGrey = Color(0xFFF2F3F8);
+  static const greyish = Color(0xFF7E7E7E);
```
Comment (vovaklh) [inline lib/presentation/themes/src/constants.dart:22]: Pls rename

File: lib/presentation/themes/src/constants.dart [hunk 29]
```diff
@@ -19,18 +19,27 @@ abstract class AppColors {
...
 
+  static const red = Color(0xFFE74C3C);
+
   static const indigo = Color(0xFF3F51B5);
   static const lightIndigo = Color(0xff6E58FC);
+
   static const lighterIndigo = Color(0xFF9E8FFD);
+  static const extremelyLightIndigo = Color(0xFFF7F7FF);
   static const darkIndigo = Color(0xFF2E3A59);
+  static const greyishIndigo = Color(0xFF464A53);
```
Comment (vovaklh) [inline lib/presentation/themes/src/constants.dart:40]: Pls rename

File: lib/presentation/themes/src/theme_colors.dart [hunk 30]
```diff
@@ -51,6 +51,17 @@ class ThemeColors extends ThemeExtension<ThemeColors> {
   final Color termsButtonFillColor;
   final Color directionsJoinButtonBorder;
   final Color productsScaffoldBackground;
+  final Color dashboardSideMenuSelectedItem;
+  final Color dashboardSideMenuUnselectedItem;
+  final Color background;
+  final Color onBackground;
```
Comment (vovaklh) [inline lib/presentation/themes/src/theme_colors.dart:57]: The on is associated with callback. Pls rename

File: lib/presentation/themes/src/theme_colors.dart [hunk 31]
```diff
@@ -51,6 +51,17 @@ class ThemeColors extends ThemeExtension<ThemeColors> {
...
+  final Color primary;
```
Comment (vovaklh) [inline lib/presentation/themes/src/theme_colors.dart:58]: Make primary for this page

File: lib/presentation/themes/src/theme_colors.dart [hunk 32]
```diff
@@ -51,6 +51,17 @@ class ThemeColors extends ThemeExtension<ThemeColors> {
...
+  final Color primaryVariant;
```
Comment (vovaklh) [inline lib/presentation/themes/src/theme_colors.dart:59]: Rename pls

File: lib/presentation/themes/src/theme_colors.dart [hunk 33]
```diff
@@ -51,6 +51,17 @@ class ThemeColors extends ThemeExtension<ThemeColors> {
...
+  final Color secondary;
+  final Color secondaryVariant;
```
Comment (vovaklh) [inline lib/presentation/themes/src/theme_colors.dart:61]: Associate with page or widget

File: lib/presentation/themes/src/theme_text_styles.dart [hunk 34]
```diff
@@ -106,6 +106,15 @@ class ThemeTextStyles extends ThemeExtension<ThemeTextStyles> {
   final TextStyle directionsBenefitsTitle;
   final TextStyle directionsBenefitsDescription;
   final TextStyle directionsBenefitsSubtitle;
+  final TextStyle dashboardSideMenuSelectedLable;
+  final TextStyle dashboardSideMenuUnselectedLable;
+  final TextStyle allRightsReserved;
+  final TextStyle body;
+  final TextStyle inverseBody;
```
Comment (vovaklh) [inline lib/presentation/themes/src/theme_text_styles.dart:113]: Associate with page or widget

File: lib/presentation/widgets/dashboard_header.dart [hunk 35]
```diff
@@ -0,0 +1,362 @@
...
+                  ? Image.asset(
+                      AppAssets.auroraLogo,
+                      height: 68,
+                    )
+                  : const SizedBox(),
+              menu.isCollapsed
+                  ? const SizedBox()
+                  : SizedBox(width: context.screenWidth * 0.02),
+              IconButton(
+                splashColor: context.color.transparent,
```
Comment (vovaklh) [inline lib/presentation/widgets/dashboard_header.dart:61]: Pls rename transparent color in styles. Associate with page...

File: lib/presentation/widgets/dashboard_header.dart [hunk 36]
```diff
@@ -0,0 +1,362 @@
...
+  @override
+  void didChangeDependencies() {
+    removeOverlay();
+    super.didChangeDependencies();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final userNameStyle =
+        context.text.body.copyWith(fontSize: 16, fontWeight: FontWeight.bold);
```
Comment (vovaklh) [inline lib/presentation/widgets/dashboard_header.dart:174]: Pls delete variable and use directly

**Review by vovaklh — CHANGES_REQUESTED** [review]

File: lib/presentation/pages/client_profile_page/sections/bots_tab.dart (hunk 26 above)
Comment (vitalii-petrun) [reply lib/presentation/pages/client_profile_page/sections/bots_tab.dart:321]: image

**Review by vovaklh — APPROVED** [review]

**Pull request merged**